mongodbatlas/resource_mongodbatlas_cluster_test.go

```diff
@@ -12,51 +12,226 @@ import (
-					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
-					resource.TestCheckResourceAttr(resourceName, "name", name),
-					resource.TestCheckResourceAttr(resourceName, "disk_size_gb", "100"),
-					resource.TestCheckResourceAttrSet(resourceName, "mongo_uri"),
-					resource.TestCheckResourceAttrSet(resourceName, "replication_specs.#"),
-					resource.TestCheckResourceAttrSet(resourceName, "replication_specs.0.regions_config.#"),
-				),
-			},
-		},
-	})
+func TestAccResourceMongoDBAtlasCluster_basicAzure(t *testing.T) {
+	var cluster matlas.Cluster
+
+	resourceName := "mongodbatlas_cluster.test"
+	projectID := os.Getenv("MONGODB_ATLAS_PROJECT_ID")
+	var region string = "US_EAST_2"
+
+	for tier, size := range matlas.DefaultDiskSizeGB["AZURE"] {
+		name := fmt.Sprintf("testAcc-%s-%s-%ggb-%s", "AZURE", tier, size, acctest.RandString(1))
+
+		t.Run(name, func(t *testing.T) {
+			resource.ParallelTest(t, resource.TestCase{
+				PreCheck:     func() { testAccPreCheck(t) },
+				Providers:    testAccProviders,
+				CheckDestroy: testAccCheckMongoDBAtlasClusterDestroy,
+				Steps: []resource.TestStep{
+					{
+						Config: testAccMongoDBAtlasClusterConfigAzure(projectID, name, "true", tier, region),
+						Check: resource.ComposeTestCheckFunc(
+							testAccCheckMongoDBAtlasClusterExists(resourceName, &cluster),
+							testAccCheckMongoDBAtlasClusterAttributes(&cluster, name),
+							resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+							resource.TestCheckResourceAttr(resourceName, "name", name),
+							resource.TestCheckResourceAttrSet(resourceName, "mongo_uri"),
+							resource.TestCheckResourceAttrSet(resourceName, "replication_specs.#"),
+							resource.TestCheckResourceAttrSet(resourceName, "replication_specs.0.regions_config.#"),
+						),
+					},
+					{
+						Config: testAccMongoDBAtlasClusterConfigAzure(projectID, name, "false", tier, region),
+						Check: resource.ComposeTestCheckFunc(
+							testAccCheckMongoDBAtlasClusterExists(resourceName, &cluster),
+							testAccCheckMongoDBAtlasClusterAttributes(&cluster, name),
+							resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+							resource.TestCheckResourceAttr(resourceName, "name", name),
+							resource.TestCheckResourceAttrSet(resourceName, "mongo_uri"),
+							resource.TestCheckResourceAttrSet(resourceName, "replication_specs.#"),
+							resource.TestCheckResourceAttrSet(resourceName, "replication_specs.0.regions_config.#"),
+						),
+					},
+				},
+			})
+		})
+	}
 
 }
+func TestAccResourceMongoDBAtlasCluster_basicGCP(t *testing.T) {
+	var cluster matlas.Cluster
+
+	resourceName := "mongodbatlas_cluster.test"
+	projectID := os.Getenv("MONGODB_ATLAS_PROJECT_ID")
+	var region string
+
+	for tier, size := range matlas.DefaultDiskSizeGB["GCP"] {
+		name := fmt.Sprintf("testAcc-%s-%s-%ggb-%s", "GCP", tier, size, acctest.RandString(1))
+
+		switch tier {
+		case "M200":
```

Comment:
These are really expensive.  Can we test lower?

Reply:
It tests the lower tiers, but this switch statement only waits for each case and then sets the specific region for each one, I can remove them if you want

Reply:
So it's more just a random - if we test w/ this level we try a different region?  I'm just not sure why M200/M300 specifically since both are available in US_EAST_4?  I'm guessing I'm missing something?

Reply:
just to making a test with a different region

Reply:
Makes sense - we can use these as long as the test immediately cleans them up, if not probably best to top at at a M30 or M40.